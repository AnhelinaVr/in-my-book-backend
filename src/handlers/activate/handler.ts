import UserService from '../../services/user';
import BadRequest from '../../exceptions/bad-request';

type Body = {
  message: string;
};

const handler = async (link: string): Promise<Body> => {
  try {
    if (!link) {
      throw new BadRequest({
        message: "Can't activate user without activation link",
      });
    }

    await UserService.activate(link);

    return { message: 'User activated' };
  } catch (error) {
    throw new BadRequest({
      message: error instanceof Error ? error?.message : undefined,
    });
  }
};

export default handler;
