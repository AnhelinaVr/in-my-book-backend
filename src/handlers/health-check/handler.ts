export const handlerName = 'health-check';

const handler = async (): Promise<string> => 'ok';

export default handler;
