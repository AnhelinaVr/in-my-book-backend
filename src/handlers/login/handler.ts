import BadRequest from '../../exceptions/bad-request';
import UserService from '../../services/user';

type Params = {
  email: string;
  password: string;
};
type Body = Awaited<ReturnType<typeof UserService.login>>;

const handler = async ({ email, password }: Params): Promise<Body> => {
  try {
    if (!email || !password) {
      throw new BadRequest({
        message: "Can't login without email or password",
      });
    }

    const user = await UserService.login(email, password);

    return user;
  } catch (error) {
    throw new BadRequest({
      message: error instanceof Error ? error?.message : undefined,
    });
  }
};

export default handler;
