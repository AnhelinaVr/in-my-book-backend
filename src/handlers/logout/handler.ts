import UserService from '../../services/user';
import BadRequest from '../../exceptions/bad-request';

const handler = async (refreshToken: string): Promise<void> => {
  try {
    if (!refreshToken) {
      throw new BadRequest({
        message: 'User is already logged out',
      });
    }

    await UserService.logout(refreshToken);
  } catch (error) {
    throw new BadRequest({
      message: error instanceof Error ? error?.message : undefined,
    });
  }
};

export default handler;
