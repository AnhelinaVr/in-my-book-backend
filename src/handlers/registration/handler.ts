import UserService from '../../services/user';
import BadRequest from '../../exceptions/bad-request';

type Params = {
  nickname: string;
  email: string;
  password: string;
};
type Result = Awaited<ReturnType<typeof UserService.registration>>;

const handler = async ({
  nickname,
  email,
  password,
}: Params): Promise<Result> => {
  try {
    if (!nickname || !email || !password) {
      throw new BadRequest({
        message: "Can't create user. Missing required fields",
      });
    }

    const user = await UserService.registration(nickname, email, password);
    console.log('user', JSON.stringify(user));
    return user;
  } catch (error) {
    throw new BadRequest({
      message: error instanceof Error ? error?.message : undefined,
    });
  }
};

export default handler;
