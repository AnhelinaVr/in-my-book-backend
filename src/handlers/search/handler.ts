/* 
request:
query: {
  author: string // author name
  query: string // both authors and name
  page: number
  limit: number
  sort: {
    field: enum [year, name, author, rating]
    direction: desc/asc, a-z/z-a
  }
}
/search?query=qqq&page=0&sort[direction]=desc&sort[field]=name&limit=20
*/
