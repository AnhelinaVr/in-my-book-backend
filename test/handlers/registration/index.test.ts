import handler from '../../../src/handlers/registration/handler';
import { fixtures } from './fixtures';

describe('Handler - registration - HTTP', () => {
  it('should return response', async () => {
    const result = await handler(fixtures);

    expect(result).toMatchSnapshot();
  });
});
