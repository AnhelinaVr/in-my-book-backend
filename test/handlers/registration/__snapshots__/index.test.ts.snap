// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Handler - registration - HTTP should return response: handler result 1`] = `
{
  "tokens": {
    "access": "token",
    "refresh": "token",
  },
  "user": {
    "email": "test@test.com",
    "id": "test-id",
    "isActivated": true,
    "nickname": "TestNickname",
  },
}
`;

exports[`Handler - registration - HTTP should return response: send mail calls 1`] = `
[
  [
    {
      "from": "test.in.my.book@outlook.com",
      "html": "
<div>
    <h1>To activate follow the link</h1>
    <a href="test-api-url/activate/test-id-1">test-api-url/activate/test-id-1</a>
</div>",
      "subject": "Registration on test-api-url",
      "text": "",
      "to": "test@test.com",
    },
  ],
]
`;
